### ISSUE TITLE
File->upload does not use title from project file (in server backed mode)

### ISSUE THREAD

**Issue by benjiwheeler** [LLK/scratch-gui#4145]
This is a holdover from https://github.com/LLK/scratch-gui/issues/4123 , which was partially resolved by https://github.com/LLK/scratch-gui/pull/4134 .

### Expected Behavior

1. You're logged in, editing your own project
2. You do file->upload
3. You see your new project, with a new url! great!
4. You expect the title to match the filename of your local file

### Actual Behavior

Project title is generic: 'Untitled-5'

### Discussion

The problem is a bit tricky...

Basically the project-state goes through the states:

loading upload ->
showing without id ->
creating new
It's at that creating new step that we want to set the new title. The problem is that from that point, you can look back one state (using prevprops) easily, but there’s no immediately obvious way in project-saver-hoc to say “oh, we’re creating a new project because of an upload, it’s not a new default project”.

**Comment by paulkaplan** [comment]
Edited the title, but wanted to note that I believe this only impacts the server backed mode, and does not cause a regression with the Desktop / standalone mode

**Comment by benjiwheeler** [comment]
Thanks for clarifying @paulkaplan, I agree

**Comment by benjiwheeler** [comment]
This was resolved by https://github.com/LLK/scratch-gui/pull/4700 , which changed uploading to use the current project id.

However, note that if we ever move to a UX where the user is able to choose between upload-in-place and upload-to-new-file, we'll want to revisit this.
